src/resize-sensor/ResizeSensor.js

```diff
@@ -1,51 +1,42 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import { props, t, pure } from '../utils';
+import { props, t } from '../utils';
 import _ElementQueries from 'css-element-queries/src/ElementQueries';
 import _ResizeSensor from 'css-element-queries/src/ResizeSensor';
 
-@pure
 @props({
   children: t.ReactChildren,
   onResize: t.Function
 })
 export default class ResizeSensor extends React.Component {
 
   attachResizeSensor = () => {
-    this.resizeSensor = new _ResizeSensor(
-      ReactDOM.findDOMNode(this),
-      this.onResize
-    );
-  }
-
-  detachResizeSensor = () => {
-    this.resizeSensor.detach(ReactDOM.findDOMNode(this));
-  }
-
-  updateResizeSensor = () => {
-    if (this.resizeSensor) {
-      this.detachResizeSensor();
+    const element = ReactDOM.findDOMNode(this);
+    if (!element.resizedAttached) {
+      this.resizeSensor = new _ResizeSensor(element, this.onResize);
     }
-    this.attachResizeSensor();
   }
 
   componentDidMount() {
     _ElementQueries.listen();
-    this.updateResizeSensor();
+    this.attachResizeSensor();
   }
 
   componentDidUpdate() {
-    this.updateResizeSensor();
-  }
-
-  componentWillUnmount() {
```

Comment:
nope, `Button` + `TextOverflow` throws an error for the reason I wrote above... If you agree I would leave it as it is and, if we have performance issues, find a way to determine if contains or not the resize sensor element (I wanted to avoid this as it would need a css query selector which we'll then need to keep up to date)